src/uct/ib/base/ib_device.c

```diff
@@ -646,6 +646,41 @@ uint8_t uct_ib_to_fabric_time(double time)
     }
 }
 
+uint8_t uct_ib_to_rnr_fabric_time(double time)
+{
+    const uint8_t max_rnr = 32;
+    double rnr_time_ms[max_rnr];
+    double avg_ms;
+    uint8_t t;
+
+    for (t = 0; t < max_rnr; t++) {
+        if (t >= 4) {
+            if (ucs_is_even(t + 1)) {
+                rnr_time_ms[t] = rnr_time_ms[t - 1] + rnr_time_ms[t - 4];
+            } else {
+                rnr_time_ms[t] = rnr_time_ms[t - 1] + rnr_time_ms[t - 3];
+            }
+        } else {
+            rnr_time_ms[t] = 0.01 * (t + 1);
+        }
+
+        if (time * UCS_MSEC_PER_SEC <= rnr_time_ms[t]) {
```

Comment:
`(time * UCS_MSEC_PER_SEC)`

Reply:
replaced by `time_ms` that's `time * UCS_MSEC_PER_SEC`
wdyt?